controllers/storagecluster/cephcluster.go

```diff
@@ -708,15 +713,13 @@ func getMonCount(monCount int, arbiter bool) int {
 			return count
 		}
 	}
-
-	if arbiter {
-		return defaults.ArbiterModeMonCount
-	}
-
+	monCount := sc.Spec.ManagedResources.CephCluster.MonCount
 	if monCount != 0 {
 		return monCount
 	}
-
+	if arbiterEnabled(sc) {
```

Comment:
Let's check for `arbiterEnabled()` before the custom setting. In stretch mode there must be exactly 5 mons and they can't override it.